management/src/test/scala/akka/management/HealthChecksSpec.scala

```diff
@@ -206,5 +211,23 @@ class HealthChecksSpec
         HealthChecks(eas, settings(checks, checks))
       }.getCause shouldEqual ctxException
     }
+    "be possible to define via ActorSystem Setup" in {
+      val readinessSetup = ReadinessCheckSetup(system => List(new Ok(system), new False(system)))
+      val livenessSetup = LivenessCheckSetup(system => List(new False(system)))
+      // bootstrapSetup is needed for config (otherwise default config)
+      val bootstrapSetup = BootstrapSetup(ConfigFactory.parseString("some=thing"))
+      val actorSystemSetup = ActorSystemSetup(bootstrapSetup, readinessSetup, livenessSetup)
```

Comment:
@marcospereira @renatocaval This shows how health checks defined via ActorSystem Setup can be passed in when creating the ActorSystem. That might be a way to support dependency injection in the health checks?

Note that such Setup is used in Lagom for the PlayJson serializers:
https://github.com/lagom/lagom/blob/master/service/scaladsl/server/src/main/scala/com/lightbend/lagom/scaladsl/server/LagomApplicationLoader.scala#L261-L268

Reply:
This looks great!

I think this will also require some basic infrastructure in Play/Lagom. Which is ok, but probably won't make it in the 2.7.0/1.5.0 releases. 

We can add it later and we don't need to be in a hurry to have it ready right now in akka-management.